sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferClient.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.work.Constraints;
+import androidx.work.Data;
+import androidx.work.ExistingWorkPolicy;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import com.azure.android.storage.blob.StorageBlobClient;
+import com.azure.android.storage.blob.credentials.SasTokenCredential;
+import com.azure.android.storage.blob.interceptor.SasTokenCredentialInterceptor;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * A type that exposes APIs for blob transfer.
+ */
+public class TransferClient {
+    private static final String TAG = TransferClient.class.getSimpleName();
+    private final Context context;
+    private final TransferDatabase db;
+    private final StorageBlobClient blobClient;
+
+    /**
+     * Creates a {@link TransferClient} that uses provided {@link StorageBlobClient}
+     * for transfers.
+     *
+     * @param context the context
+     * @param blobClient the blob storage client
+     */
+    public TransferClient(Context context, StorageBlobClient blobClient) {
+        this.context = context;
+        this.db = TransferDatabase.get(context);
+        this.blobClient = blobClient;
+    }
+
+    /**
+     * Creates a {@link TransferClient} that uses provided blob storage endpoint and SAS token.
+     *
+     * @param context the context
+     * @param storageUrl the blob storage url
+     * @param sasToken the SAS token
+     */
+    public TransferClient(Context context, String storageUrl, String sasToken) {
+        this(context,
+            new StorageBlobClient.Builder()
+                .setBlobServiceUrl(storageUrl)
+                .setCredentialInterceptor(new SasTokenCredentialInterceptor(new SasTokenCredential(sasToken)))
```

Comment:
I may have missed it, but where/how does the customer provide the method that will be called to retrieve a new SAS token / preauthenticated URL when this token expires?

Reply:
your concern is correct, the `SasTokenCredential` type is only support const final SAS token as of today. Mutation of SAS token via callback is yet to define, that will be separate PR.